system/include/emscripten/val.h

```diff
@@ -126,7 +126,7 @@ namespace emscripten {
             */
 
             static EM_METHOD_CALLER get_method_caller() {
```

Comment:
I don't think this even needs to be a function anymore - thread-local is lazily initialized on access anyway, so you could unwrap it.

Reply:
Do you mean to pull `thread_local EM_METHOD_CALLER mc` out as a member variable instead of declaring within `get_method_caller` or merging `get_method_caller`/`init_method_caller` into a single method with `mc` still being local to that function scope?

Reply:
I was thinking of extracting variable out of function scope, but merging also sounds good.